wsmaster/che-core-api-machine-shared/src/main/java/org/eclipse/che/api/machine/shared/dto/MachineSourceDto.java

```diff
@@ -30,10 +31,33 @@
     MachineSourceDto withType(String type);
 
     @Override
-    @FactoryParameter(obligation = MANDATORY)
+    @FactoryParameter(obligation = OPTIONAL)
```

Comment:
We don't have a good way to handle such parameters in a factory now. Yes, I saw check in WSservice, but I'm saying about saving of factory, not accepting.
Probably we need some new features in factory validation framework to validate different types of implementation specific configs.
So I think it is ok for now